### PR TITLE
Add Support for API report

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
Pruned DTS file: https://gist.github.com/schmidt-sebastian/2401be010ce057988495823ffe446a1d

### CODE REVIEW

File: packages/database/scripts/api-report.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+// TODOD(mrschmidt): Combine this script with the Firestore version.
+
+/* eslint-disable no-console */
+
+// This script takes the output of the API Extractor, post-processes it using
+// the pruned-dts script and then invokes API report to generate a report
+// that only includes exported symbols. This is all done in temporary folders,
+// all configuration is auto-generated for each run.
+
+const baseApiExtractorConfigFile: string = path.resolve(
```
Comment (Feiyang1) [inline packages/database/scripts/api-report.ts:38]: Seems we should share this script across projects, but let's do it in a separate PR.

**Review by Feiyang1 — APPROVED** [review]

File: packages/database/scripts/api-report.ts (hunk 1 above)
Comment (schmidt-sebastian) [reply packages/database/scripts/api-report.ts:38]: Yup, I thought I addressed this task by adding the TODO :)
